test/index/functions/operators.snapshot.cc

```diff
@@ -0,0 +1,265 @@
+  // extra-args: -std=c++23
+  
+  // Overloaded operators
+  
+  struct MyStream {};
+//       ^^^^^^^^ definition [..] MyStream#
+  
+  MyStream &operator<<(MyStream &s, int) { return s; }
+//^^^^^^^^ reference [..] MyStream#
+//          ^^^^^^^^ definition [..] `operator<<`(97f6638197cf8bc4).
+//                     ^^^^^^^^ reference [..] MyStream#
+//                               ^ definition local 0
+//                                                ^ reference local 0
+  MyStream &operator<<(MyStream &s, const char *) { return s; }
+//^^^^^^^^ reference [..] MyStream#
+//          ^^^^^^^^ definition [..] `operator<<`(5651711ec4adbebf).
+//                     ^^^^^^^^ reference [..] MyStream#
+//                               ^ definition local 1
+//                                                         ^ reference local 1
+  
+  struct Int { int val; };
+//       ^^^ definition [..] Int#
+  struct IntPair { Int x; Int y; };
+//       ^^^^^^^ definition [..] IntPair#
+//                 ^^^ reference [..] Int#
+//                        ^^^ reference [..] Int#
+  
+  IntPair operator,(Int x, Int y) {
+//^^^^^^^ reference [..] IntPair#
+//        ^^^^^^^^ definition [..] `operator,`(56a2b3932346e301).
+//                  ^^^ reference [..] Int#
+//                      ^ definition local 2
+//                         ^^^ reference [..] Int#
+//                             ^ definition local 3
+    return IntPair{x, y};
+//         ^^^^^^^ reference [..] IntPair#
+//                 ^ reference local 2
+//                    ^ reference local 3
+  }
+  
+  Int operator++(Int &i, int) {
+//^^^ reference [..] Int#
+//    ^^^^^^^^ definition [..] operator++(be31e3af2b2ba0e).
+//               ^^^ reference [..] Int#
+//                    ^ definition local 4
+    return Int{i.val+1};
+//         ^^^ reference [..] Int#
+//             ^ reference local 4
+  }
+  
+  struct FnLike {
+//       ^^^^^^ definition [..] FnLike#
+    int operator()() { return 0; }
+//      ^^^^^^^^ definition [..] FnLike#`operator()`(b126dc7c1de90089).
+  };
+  
+  struct Table {
+//       ^^^^^ definition [..] Table#
+    int value;
+    Table *operator->() { return this; }
+//  ^^^^^ reference [..] Table#
+//         ^^^^^^^^ definition [..] Table#`operator->`(ed921902444779f1).
+    int &operator[](int i, int j) { // since C++23
+//       ^^^^^^^^ definition [..] Table#`operator[]`(77cf9b7ed2f5124c).
+//                      ^ definition local 5
+//                             ^ definition local 6
+      return this->value;
```

Comment:
TODO: Missing a reference to `operator->`.

Reply:
Looks like I was wrong, when the receiver is of pointer type, it doesn't go through `->`.